src/domain/challenge/challenge/challenge.service.ts

```diff
@@ -67,55 +54,38 @@ export class ChallengeService {
     if (!challenge.context) {
       challenge.context = new Context();
     }
+
+    if (!challenge.community) {
+      challenge.community = new Community(
+        challenge.name,
+        CommunityType.CHALLENGE,
+        [AuthorizationRoles.Members]
```

Comment:
There is an implicit mapping Roles <--> RestrictedGroupNames. I'd rather have Roles in the Community entity, logically that's what we have, not RestrictedGroupNames.

Reply:
Pick up as a separate topic